react-native-aztec/android/src/main/java/org/wordpress/mobile/ReactNativeAztec/ReactAztecText.java

```diff
@@ -525,11 +547,19 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
 
         @Override
         public void afterTextChanged(Editable s) {
-            if (!mIsSettingTextFromJS && mListeners != null) {
+            if (shouldRedirectTextChangeCalls()) {
                 for (TextWatcher listener : mListeners) {
                     listener.afterTextChanged(s);
                 }
             }
         }
     }
+
+    private boolean shouldRedirectTextChangeCalls() {
+        return (!mIsSettingTextFromJS || isPreTag()) && mListeners != null;
```

Comment:
The conditional has now started to be harder to read/follow. Can we split it up into documented individual ones that return early? WDYT?

For example:
```
        if (mListeners == null) {
            // No listeners so, no-one to delegate the calls to
            return false;
        }
```

Reply:
Good suggestion. I have split condition into multiple individuals.